src/sentry/silo/client.py

```diff
@@ -107,6 +120,26 @@ def request(
         return client_response
 
 
+def validate_region_ip_address(ip: str) -> bool:
+    """
+    Checks if the provided IP address is a Region Silo IP address.
+    """
+    if not ALLOWED_REGION_IP_ADDRESSES:
+        sentry_sdk.capture_exception(
+            RegionResolutionError(f"Disallowed Region Silo IP address: {ip}")
```

Comment:
Should this case and the disallowed case have different messages? Knowing that the allowed list is empty could help with debugging.